vars/neoDeploy.groovy

```diff
@@ -8,222 +11,218 @@ import com.sap.piper.tools.ToolDescriptor
             }
-        }
 
-        if (deployMode == 'warParams') {
-            applicationName = utils.getMandatoryParameter(configuration, 'applicationName')
-            runtime = utils.getMandatoryParameter(configuration, 'runtime')
-            runtimeVersion = utils.getMandatoryParameter(configuration, 'runtimeVersion')
-            def vmSizes = ['lite', 'pro', 'prem', 'prem-plus']
-            vmSize = configuration.vmSize
-            if (! (vmSize in vmSizes)) {
-                throw new Exception("[neoDeploy] Invalid vmSize = '${vmSize}'. Valid 'vmSize' values are: ${vmSizes}.")
+            echo "Link to the application dashboard: ${neoCommandHelper.cloudCockpitLink()}"
+
+            if (warAction == 'rolling-update') {
+                sh neoCommandHelper.rollingUpdateCommand()
+            } else {
+                sh neoCommandHelper.deployCommand()
+                sh neoCommandHelper.restartCommand()
             }
-        }
 
-        if (deployMode in ['mta','warParams']) {
-            deployHost = utils.getMandatoryParameter(configuration, 'host')
-            deployAccount = utils.getMandatoryParameter(configuration, 'account')
-        }
 
-        def neo = new ToolDescriptor('SAP Cloud Platform Console Client', 'NEO_HOME', 'neoHome', '/tools/', 'neo.sh', null, 'version')
-        def neoExecutable = neo.getToolExecutable(this, configuration)
-        def neoDeployScript = """#!/bin/bash
-                                 "${neoExecutable}" ${warAction} \
-                                 --source "${archivePath}" \
-                              """
-
-        if (deployMode in ['mta', 'warParams']) {
-            neoDeployScript +=
-                    """--host '${deployHost}' \
-                    --account '${deployAccount}' \
-                    """
-        }
+        } else if (deployMode == 'mta') {
+            warAction = 'deploy-mta'
 
-        if (deployMode == 'mta') {
-            neoDeployScript += "--synchronous"
+            sh neoCommandHelper.deployMta()
         }
+    }
+    catch (Exception ex) {
+        echo "Error while deploying to SAP Cloud Platform. Here are the neo.sh logs:"
+        sh "cat ${neoToolDescriptor.getToolLocation(script, configuration)}/tools/log/*"
+        throw ex
+    }
+}
 
-        if (deployMode == 'warParams') {
-            neoDeployScript +=
-                    """--application '${applicationName}' \
-                    --runtime '${runtime}' \
-                    --runtime-version '${runtimeVersion}' \
-                    --size '${vmSize}'"""
-        }
+private boolean isAppRunning(NeoCommandHelper commandHelper) {
+    def status = sh script: "${commandHelper.statusCommand()} || true", returnStdout: true
+    return status.contains('Status: STARTED')
+}
 
-        if (deployMode == 'warPropertiesFile') {
-            neoDeployScript +=
-                    """${propertiesFile}"""
-        }
+private handleCompatibility(script, parameters) {
```

Comment:
Since we announed incompatible changes for the next release: would be possible to put the compatibilty handing to the trash bin?

Reply:
Already now? Shouldn't we give the users time to adopt or stick to a stable version?

Reply:
my proposal is: wait with merging (I guess it will take some time anyway) and cleanup it now. This is cheaper since there is no need to e.g. adjust tests which can be removed afterwards since the code has less bells and whistles compared to the current status. More opinions from other colleagues welcome.

Reply:
My hope is to get this merged until end of our sprint (Tuesday). 
Tests are already adjusted to the new format. We can just delete the tests which are there to test the old configuration way in the follow-up.